android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/annotation/core/pdf/PdfUtil.kt

```diff
@@ -0,0 +1,367 @@
+package com.github.barteksc.pdfviewer.annotation.core.pdf
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.graphics.PointF
+import android.graphics.pdf.PdfRenderer
+import android.os.ParcelFileDescriptor
+import com.github.barteksc.pdfviewer.annotation.core.annotations.Annotation
+import com.github.barteksc.pdfviewer.annotation.core.annotations.AnnotationManager
+import com.github.barteksc.pdfviewer.annotation.core.annotations.AnnotationType
+import com.github.barteksc.pdfviewer.annotation.core.annotations.toCircleShape
+import com.github.barteksc.pdfviewer.annotation.core.annotations.toRectangleShape
+import com.github.barteksc.pdfviewer.annotation.core.shapes.Documentation
+import com.github.barteksc.pdfviewer.annotation.core.shapes.Rectangle
+import com.github.barteksc.pdfviewer.annotation.core.shapes.Relations
+import com.github.barteksc.pdfviewer.annotation.core.shapes.Shape
+import com.github.barteksc.pdfviewer.annotation.core.shapes.ShapeType
+import com.github.barteksc.pdfviewer.annotation.core.shapes.generateRectangleCoordinates
+import com.github.barteksc.pdfviewer.annotation.core.shapes.mapJsonStringToPdfShapes
+import com.github.barteksc.pdfviewer.annotation.core.shapes.mapPdfShapesToJsonString
+import com.github.barteksc.pdfviewer.annotation.core.shapes.toAnnotation
+import com.github.barteksc.pdfviewer.util.logDebug
+import com.lowagie.text.pdf.PdfArray
+import com.lowagie.text.pdf.PdfDictionary
+import com.lowagie.text.pdf.PdfName
+import com.lowagie.text.pdf.PdfNumber
+import com.lowagie.text.pdf.PdfReader
+import com.lowagie.text.pdf.PdfString
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.InputStream
+import kotlin.properties.Delegates
+
+object PdfUtil {
+
+    private val TAG: String = PdfUtil.javaClass.simpleName
+
+    /** Extract PDF extras from the annotations for the given PDF file path.
+     * Check this when linking entities to annotation */
+    @Throws(IOException::class)
+    @JvmStatic
+    private fun getAnnotationsExtra(filePath: String) {
+        logDebug(TAG, "file path is: $filePath")
+
+        if (filePath.isEmpty()) throw java.lang.Exception("Input file is empty")
+        val file = File(filePath)
+        if (!file.exists()) throw java.lang.Exception("Input file does not exists")
+
+        // input stream from file
+        val inputStream: InputStream = FileInputStream(file)
+
+        // we create a reader for a certain document
+        val reader = PdfReader(inputStream)
+
+        val n = reader.numberOfPages
+
```

Comment:
Please remove white lines and redundant comments also make sure to use proper variable naming.

This `val n = reader.numberOfPages` needs renaming for example